synapse/http/federation/matrix_federation_agent.py

```diff
@@ -287,7 +313,62 @@ def _get_well_known(self, server_name):
                 raise Exception("Missing key 'm.server'")
         except Exception as e:
             raise Exception("invalid .well-known response from %s: %s" % (uri_str, e,))
-        defer.returnValue(parsed_body["m.server"].encode("ascii"))
+
+        result = parsed_body["m.server"].encode("ascii")
+
+        cache_period = _cache_period_from_headers(
+            response.headers,
+            time_now=self._reactor.seconds,
+        )
+        if cache_period is None:
+            cache_period = WELL_KNOWN_DEFAULT_CACHE_PERIOD
+        else:
+            cache_period = min(cache_period, WELL_KNOWN_MAX_CACHE_PERIOD)
+
```

Comment:
Don't we also want a min cache period here here?

Reply:
Not sure, tbh. What would you set it to? If someone says "don't cache", I'm minded to respect that.

Reply:
Oh, I hallucinated you adding a `MIN` constant. I don't have strong feelings either way, though I wouldn't be surprised if people used `max_age=0`, but we can revisit it later if it proves a problem